### DISCUSSION TITLE
Is it possible to feather only when necessary?

### DISCUSSION THREAD

**Discussion by rmccrystal** [emilk/egui#1909] [General]
This question may not be accurate to how feathering works, but I've been experimenting with egui and I've noticed that feathering seems to make the UI slightly blurry. For example, compare these two screenshots:

![image](https://user-images.githubusercontent.com/17937856/184474039-c7c8abb3-77fb-4129-b983-aff738368637.png)

![image](https://user-images.githubusercontent.com/17937856/184474031-361c40ac-0a09-4be2-ae52-cc9aff1a0b68.png)

I much prefer the look of the second screenshot.

My goal is to have a border on each widget with a width of exactly one pixel, but this doesn't seem to be achievable with feathering enabled. Even when drawing shapes that are aligned to pixels, they appear blurry like in the first screenshot. The issue with disabling feathering entirely is when rendering shapes like circles or checkboxes, they appear jagged. Is it possible to effectively disable feathering for pixel alined shapes but keep it enabled when rendering round or non pixel aligned objects?

**Comment by emilk** [comment]
No, right now feathering is completely on or off.

Perhaps you can tweak the feather size to find a better tradeoff?
